### PR TITLE
Use the latest version of yeoman-plugin

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: Vitaliy Guliy <vgulyy@redhat.com>

### What does this PR do?
Includes the latest version of `@theia/yeoman-plugin` to Che-Theia build.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16354

### CODE REVIEW

File: dockerfiles/theia/build.sh [hunk 1]
```diff
@@ -22,9 +22,9 @@ tar rf ${DIR}/asset-che-theia.tar .git-che-theia-sha1
 gzip -f ${DIR}/asset-che-theia.tar
 
 # Download plugins
-THEIA_YEOMAN_PLUGIN="${DIR}/asset-untagged-c11870b25a17d20bb7a7-theia_yeoman_plugin.theia"
+THEIA_YEOMAN_PLUGIN="${DIR}/asset-untagged-8a7963262e021dab8ae0-theia_yeoman_plugin.theia"
 if [ ! -f "${THEIA_YEOMAN_PLUGIN}" ]; then
-    curl -L -o ${THEIA_YEOMAN_PLUGIN} https://github.com/eclipse/theia-yeoman-plugin/releases/download/untagged-c11870b25a17d20bb7a7/theia_yeoman_plugin.theia
+    curl -L -o ${THEIA_YEOMAN_PLUGIN} https://github.com/eclipse/theia-yeoman-plugin/releases/download/untagged-8a7963262e021dab8ae0/theia_yeoman_plugin.theia
```
Comment (azatsarynnyy) [inline dockerfiles/theia/build.sh]: fyi @nickboldt 
I'm not sure, maybe it affects CRW built

Comment (vitaliy-guliy) [reply dockerfiles/theia/build.sh]: It's just another file

Comment (nickboldt) [reply dockerfiles/theia/build.sh]: should be OK in downstream but thx for heads up.

Are you cherrypicking this to 7.9.x branch too?

Comment (vitaliy-guliy) [reply dockerfiles/theia/build.sh]: If it's needed, I can do it

File: dockerfiles/theia/build.sh [hunk 2]
```diff
@@ -22,9 +22,9 @@ tar rf ${DIR}/asset-che-theia.tar .git-che-theia-sha1
 gzip -f ${DIR}/asset-che-theia.tar
 
 # Download plugins
-THEIA_YEOMAN_PLUGIN="${DIR}/asset-untagged-c11870b25a17d20bb7a7-theia_yeoman_plugin.theia"
+THEIA_YEOMAN_PLUGIN="${DIR}/asset-untagged-8a7963262e021dab8ae0-theia_yeoman_plugin.theia"
```
Comment (nickboldt) [inline dockerfiles/theia/build.sh]: if we make the filename generic (not containing the sha) then my downstream implementation would be simpler/more generic.

File: dockerfiles/theia/build.sh (hunk 1 above)
Comment (nickboldt) [reply dockerfiles/theia/build.sh]: LMK if you decide it's important. Plesae do it before EOD on Tuesday so we can include in 7.9.2. Please also add a link to the issue on https://github.com/eclipse/che/issues/16229 so we can track it.

Comment (vitaliy-guliy) [reply dockerfiles/theia/build.sh]: I'd say yes. We fixed the generator two months ago, but still don't use it.

**Comment by nickboldt** [comment]
crw-ci-test

File: dockerfiles/theia/build.sh (hunk 2 above)
Comment (vitaliy-guliy) [reply dockerfiles/theia/build.sh]: Done

Comment (nickboldt) [reply dockerfiles/theia/build.sh]: needs the "asset-untagged-" prefix for the che-theia and crw-theia build to find it via wildcard search for `asset-*` files. I was just asking that you remove the SHA part.

Comment (vitaliy-guliy) [reply dockerfiles/theia/build.sh]: Changed again

**Comment by vitaliy-guliy** [comment]
crw-ci-test

**Comment by nickboldt** [comment]
requires additional fix in #670

**Comment by vitaliy-guliy** [comment]
@nickboldt thanks for the fix!

**Comment by nickboldt** [comment]
No problem! I'm just happy to see the happy path tests actually pass sometimes \o/

**Pull request merged**